component/http/cache.go

```diff
@@ -0,0 +1,419 @@
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"hash/crc32"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/beatlabs/patron/log"
+)
+
+type validationContext int
+
+const (
+	// validation due to normal expiry in terms of ttl setting
+	ttlValidation validationContext = iota + 1
+	// maxAgeValidation represents a validation , happening due to max-age  header requirements
+	maxAgeValidation
+	// minFreshValidation represents a validation , happening due to min-fresh  header requirements
+	minFreshValidation
+	// maxStaleValidation represents a validation , happening due to max-stale header requirements
+	maxStaleValidation
+
+	// cacheControlHeader is the header key for cache related values
```

Comment:
IMO we can skip (most of) the comments in this section because the purpose of `cacheControl*` and `cacheHeader*` directives is clearly stated in the [RFC](https://www.ietf.org/rfc/rfc2616.txt) (section14.9 Cache-Control)